product_history/models/product_product.py

```diff
@@ -0,0 +1,478 @@
+##############################################################################
+#
+#    Product - Average Consumption Module for Odoo
+#    Copyright (C) 2013-Today GRAP (http://www.grap.coop)
+#    Copyright (C) 2019-Today: La Louve (<https://cooplalouve.fr>)
+#    Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+#    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+#    @author Julien WESTE
+#    @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from datetime import date, time
+from datetime import datetime as dt
+from datetime import timedelta as td
+
+from dateutil.relativedelta import relativedelta as rd
+from odoo import models, fields, api
+from odoo.addons.queue_job.job import job
+
+old_date = date(2015, 1, 1)
```

Comment:
Okay, I will check it out.
Actually this branch is WIP right now, I just have made it installable.

Reply:
Hi @ivantodorovich 

old_date variable is used when the first time a history of a product gets created.

If a history record of a product is already there then it considers the last record of the history and fetches the date from it to create next history records, but if there is no record of a history of a product then it takes old_date as starting date to create a history of the product.

Reply:
Do you think that we can remove this behaviour?
I see no point in having it, do you?

I guess it only make sense if you install this module in a database with a lot of moves, dating previous to 01/01/2015. But that will never happen.

If you agree, proceed to remove it

Reply:
Yes, agreed.

But there should be a start date from when to create history records when there is no history, I think the better option is we should take the first stock move's create_date, and it could be different for all the products.

So I will find the 1st stock.move record of the particular product and fetch the create_date of it to start the creation of a history.

Reply:
Okay, old_date not needed. code for stock.move I explained is already there, so I will remove it.

Reply:
Awesome :)